src/cmdline_manager.ts

```diff
@@ -28,6 +28,16 @@ export class CommandLineManager implements Disposable {
     // However, bindings are expected to cause the cmdline content to change, so we use this flag to listen to those updates.
     private redrawExpected = true;
 
+    // Much effort was put into this to make sure that race conditions behave as expected.
```

Comment:
This is difficult, and I'm not fully happy with the solution. If anyone has suggestions for this, it would be appreciated. 

The core problem is that when we spam nvim shortcuts, it triggers a cmdline_show, which we then use to update the quickpick. But then, the onChanged callback is triggered, which if we use to send anything to nvim, it breaks the completions. It is not enough to compare the onChanged text with the last sent text from nvim, because onChanged can lag behind. 

But, if we debounce the onChanged, we create a problem where typing a command and rapidly using a shortcut like C-h causes nvim to operate on outdated text. 

I haven't found a robust way to handle both cases yet.